linalg/sparsemat.hpp

```diff
@@ -80,9 +86,30 @@ class SparseMatrix : public AbstractSparseMatrix
    void Destroy();   // Delete all owned data
    void SetEmpty();  // Init all entries with empty values
 
+   bool useCuSparse{true}; //Use CuSparse if available
```

Comment:
This should go inside the `#ifdef MFEM_USE_CUDA` guard.

Reply:
If we do that then users have to add #ifdef guards when they toggle cusparse in cpu builds. Like this it ends but being an noop.